react-interactive/src/Layout/VerticalSplit.tsx

```diff
@@ -0,0 +1,240 @@
+
+import { LeftArrow, RightArrow, SVGIcons } from '@gpa-gemstone/gpa-symbols';
+import * as React from 'react';
+import { number } from 'yargs';
+import SplitDrawer from './SplitDrawer';
+import SplitSection from './SplitSection';
+
+interface IProps {
+    style?: any,
+    sliderStyle?: any,
+}
+
+// Props Description:
+// style => style of the encompasing div
+
+const VerticalSplit: React.FunctionComponent<IProps> = (props) => {
+    const divRef = React.useRef<any>(null);
+
+    const [currentWidth, setCurrentWidth] = React.useState<number>(0);
+    const [sections, setSections] = React.useState<React.ReactElement<any>[]>([]);
+    const [width, setWidth] = React.useState<number[]>([]);
+
+    const [LeftDrawer, setLeftDrawer] = React.useState<React.ReactElement<any>[]>([]);
+    const [LeftDrawerWidth, setLeftDrawerWidth] = React.useState<number[]>([]);
+
+    const [activeSlider, setActiveSlider] = React.useState<number>(-1);
+    const [sliderOriginal, setSliderOriginal] = React.useState<number>(0);
+
+    React.useLayoutEffect(() => {
+        setCurrentWidth(divRef.current.offsetWidth ?? 0);
+    })
+
+    React.useEffect(() => {
+      
+        setSections(React.Children.map(props.children,(child) => {
+            if (!React.isValidElement(child))
+                return null;
+            if ((child as React.ReactElement<any>).type === SplitSection)
+                return (child as React.ReactElement<any>)
+            else
+                return null 
+
+        })?.filter(item => item !== null) ?? []);
+
+        setLeftDrawer(React.Children.map(props.children,(child) => {
+            if (!React.isValidElement(child))
+                return null;
+            if ((child as React.ReactElement<any>).type === SplitDrawer)
+                return (child as React.ReactElement<any>)
+            else
+                return null 
+
+        })?.filter(item => item !== null) ?? []);
+
+    }, [props.children])
+
+    React.useEffect(() => {
+            setLeftDrawerWidth((w) => {
+                let wupdated: number[] = [...w];
+                
+                if (w.length > LeftDrawer.length)
+                    wupdated = w.slice(undefined,LeftDrawer.length);
+                if (w.length < LeftDrawer.length)
+                    wupdated = [...w,...LeftDrawer.slice(w.length,undefined).map((s) => (s.props as any).open ? (s.props as any).width as number : 0)];
+                
+                return wupdated.map((width,index) => {
+                    if (width == 0)
+                        return 0;
+                    if (width < (LeftDrawer[index].props as any).minWidth)
+                        return (LeftDrawer[index].props as any).minWidth as number
+                    if (width > (LeftDrawer[index].props as any).maxWidth)
+                        return (LeftDrawer[index].props as any).maxWidth as number
+                    return width;
+                }) })
```

Comment:
```suggestion
                return wupdated.map((wupdatedWidth,index) => {
                    if (wupdatedWidth === 0)
                        return 0;
                    if (wupdatedWidth < (LeftDrawer[index].props as any).minWidth)
                        return (LeftDrawer[index].props as any).minWidth as number
                    if (wupdatedWidth > (LeftDrawer[index].props as any).maxWidth)
                        return (LeftDrawer[index].props as any).maxWidth as number
                    return wupdatedWidth;
                }) })
```

Fix for lint errors, triple equals and variable shadowing.